### DISCUSSION TITLE
Is it possible to create new feature flags using TypeScript?

### DISCUSSION THREAD

**Discussion by carrie-shepherd** [featurehub-io/featurehub#1059] [Q&A]
Hi

We are looking to automate the creation of feature flags for an application.
I have looked for resources to see if it is possible to do this with TypeScript, but I have not found anything.

Is this possible to do?

Thanks.

Caroline

**Comment by rvowles** [comment]
Hi Caroline! The APi for FeatureHub is defined in OpenAPI and we have a generated Admin Development Kit here: https://github.com/featurehub-io/featurehub-javascript-adk

We use it extensively in our testing, so you can find examples here for instance: https://github.com/featurehub-io/featurehub/blob/main/adks/e2e-sdk/app/steps/flag_steps.ts

if you look further up that folder you will see it being used to create all sorts

**Reply by carrie-shepherd** [reply]
Thank you very much.

**Reply by IrinaSouth** [reply]
@carrie-shepherd  Here is the link to the docs and OpenAPI files https://docs.featurehub.io/featurehub/latest/admin-development-kit.html#_openapi_files
